LuaUI/Widgets/cmd_hold_ctrl_to_place_and_morph.lua

```diff
@@ -0,0 +1,126 @@
+local widgetName = "Hold Ctrl during placement to morph";
+
+function widget:GetInfo()
+	return {
+		name      = widgetName,
+		version   = '1.0',
+		desc      = "Hold the Ctrl key while placing a Cornea, Aegis, Radar Tower or Geothermal Reactor to issue a morph order to the nanoframe when it's created (which will make it start morphing once finished).",
+		author    = "dunno",
+		date      = "2022-05-18",
+		license   = "MIT",
+		layer     = 0,
+		enabled   = true
+	}
+end
+
+---@alias UnitInternalName string
+---@alias UnitDefId integer
+---@alias UnitId integer
+
+---@param names UnitInternalName[]
+---@return table<UnitDefId, boolean>
+local function CreateUnitDefIdSet(names)
+	local result = {}
+	for _, name in ipairs(names) do
+		result[UnitDefNames[name].id] = true
+	end
+	return result
+end
+
+local abs = math.abs
+local spGetUnitPosition = Spring.GetUnitPosition
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+
+local myTeamID
+
+local morphableUnitDefIds = CreateUnitDefIdSet({'staticjammer', 'staticshield', 'staticradar', 'energygeo'})
+
+---@type table<UnitId,{x:number,z:number}[]>
+local buildingsToMorphByBuilder = {}
+
+function widget:Initialize()
+	myTeamID = Spring.GetMyTeamID()
+end
+
```

Comment:
Handle changing team (debugging with `/team`, comm sharing, etc):
```suggestion
widget.PlayerChanged = widget.Initialize

```